CMakeLists.txt

```diff
@@ -27,3 +31,18 @@ add_subdirectory(src)
 add_subdirectory(doc)
 add_subdirectory(test)
 add_subdirectory(cpack)
+
+if(SYSTEM_NAME STREQUAL "FreeBSD")
+  # use a linker from ports, e.g. lld50 from llvm
+  message("use a linker built from ports")
+
+  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") # using Clang
```

Comment:
require_clang_version is a no-op if run with a different compiler, so no reason to check this here.